File: 2475-number-of-unequal-triplets-in-array/2475-number-of-unequal-triplets-in-array.java

class Solution {
  public int unequalTriplets(int[] nums) {
    int ans = 0;
    int prev = 0;
    int next = nums.length;
    Map<Integer, Integer> count = new HashMap<>();

    for (final int num : nums)
      count.merge(num, 1, Integer::sum);

    for (final int freq : count.values()) {
      next -= freq;
      ans += prev * freq * next;
      prev += freq;
    }

    return ans;
  }
}

File: 2475-number-of-unequal-triplets-in-array/2475-number-of-unequal-triplets-in-array.py

class Solution:
  def unequalTriplets(self, nums: List[int]) -> int:
    ans = 0
    prev = 0
    next = len(nums)

    for freq in collections.Counter(nums).values():
      next -= freq
      ans += prev * freq * next
      prev += freq

    return ans